src/ui/backend/BaseBacking.js

```diff
@@ -16,38 +16,59 @@
 
 import setProperty from 'util/setProperty';
 
-// keys map to properties
 var BASE_STYLE_PROPS = {
-  'x': { value: 0 },
-  'y': { value: 0 },
-  // translate
-  'offsetX': { value: 0 },
-  'offsetY': { value: 0 },
-  // rotation and scale
-  'anchorX': { value: 0 },
-  'anchorY': { value: 0 },
-  'centerAnchor': { value: false },
-  'width': { cb: '_onResize' },
-  'height': { cb: '_onResize' },
-  'r': { value: 0 },
-  'opacity': { value: 1 },
-  'zIndex': {
-    value: 0,
-    cb: '_onZIndex'
-  },
-  'scale': { value: 1 },
-  'scaleX': { value: 1 },
-  'scaleY': { value: 1 },
-  'flipX': { value: false },
-  'flipY': { value: false },
-  'visible': { value: true },
-  'clip': { value: false },
-  'backgroundColor': { value: '' },
-  'compositeOperation': { value: '' }
+  x: 0,
+  y: 0,
+  offsetX: 0,
+  offsetY: 0,
+  anchorX: 0,
+  anchorY: 0,
+  centerAnchor: false,
+  _width: 0,
+  _height: 0,
+  r: 0,
+  opacity: 1,
+  _zIndex: 0,
+  scale: 1,
+  scaleX: 1,
+  scaleY: 1,
+  flipX: false,
+  flipY: false,
+  visible: true,
+  clip: false,
+  backgroundColor: '',
+  compositeOperation: ''
 };
 
 export default class BaseBacking {
 
+  constructor () {
+    for (var key in BASE_STYLE_PROPS) {
```

Comment:
I tried both and results seem identical.

It is hard to spot tiny changes of performance without a way to automate benchmarks on a running game (Ideally we need to engineer a system that can automatically run scenarios on a game. Will be good for debugging and testing performance).